### DISCUSSION TITLE
How do I execute command or send chat as player?

### DISCUSSION THREAD

**Discussion by Wolfyxon** [FabricMC/fabric-api#1489] [Mod Dev Support] [answered]
I'm working on mod that allows you to run specific commands when pressing keys. I'm making it to help players being portal banned on some servers. 
Please help I'm new to minecraft mods

**Comment by kanpov** [comment]
For making your own commands, check out [https://fabricmc.net/wiki/tutorial:commands](url)

For making keybindings, check out [https://fabricmc.net/wiki/tutorial:keybinds](url)

**Comment by nok-ko** [comment, marked as answer]
You can execute commands the same way you send chat messages! Use [`sendChatMessage`](https://maven.fabricmc.net/docs/yarn-1.17+build.13/net/minecraft/client/network/ClientPlayerEntity.html#sendChatMessage(java.lang.String)) in `ClientPlayerEntity`.

**Reply by Wolfyxon** [reply]
Thanks

**Reply by Panda98277** [reply]
It seems can't  work on 1.20.2+build.1

**Reply by AnOpenSauceDev** [reply]
This is because of the new chat reporting system. [I personally did it this way.](https://github.com/AnOpenSauceDev/archbtw/blob/e1897470a6ea92838ecd0422e5ca2fc8e536ac22/src/main/java/com/github/anopensaucedev/archbtw/mixin/ArchBTW.java#L25)
